hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/ShortCircuitRegistry.java

```diff
@@ -322,7 +322,7 @@ public NewShmInfo createNewMemorySegment(String clientName,
         shm = new RegisteredShm(clientName, shmId, fis, this);
       } finally {
         if (shm == null) {
-          IOUtils.closeQuietly(fis);
+          IOUtils.cleanupWithLogger(null, fis);
```

Comment:
why not use `closeStream(fis)`?

Reply:
Sounds good, `closeStream()` is also accessible and does the same so it's cleaner than passing null for each. Let me make this change.